extensions/amp-mustache/0.2/amp-mustache.js

```diff
@@ -133,16 +122,39 @@ export class AmpMustache extends BaseTemplate {
   /**
    *
    * @param {string} html
-   * @return {!Element}
+   * @return {!Promise<!Element>}
    * @private
    */
   purifyAndSetHtml_(html) {
-    const body = this.purifier_.purifyHtml(`<div>${html}</div>`);
-    const div = body.firstElementChild;
-    return this.unwrap(div);
+    return getServicePromiseForDoc(this.element, 'purifier').then(purifier => {
```

Comment:
While the underlying method does cache the service (so only a single promise would be resolved after the first usage), this might have a future performance increase with something that returns sync or is cached locally.

Reply:
I don't think I understand...what would return sync or cache locally? How can I improve this?
Note also that I'm switching it back to the sync `getService` and instantiating DOMPurify once per window instead of once per ampdoc.

Reply:
Think in the future we should take a look at whether `getServicePromiseForDoc(this.element, 'purifier')` needs to be async.

If sync, the promise resolution of an already sync available item can be avoided.

Not within the scope of this PR though since there are other usages throughout the codebase.